pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -77,16 +77,20 @@ func SetDefaults_Deployment(obj *Deployment) {
 			strategy.RollingUpdate = &rollingUpdate
 		}
 		if strategy.RollingUpdate.MaxUnavailable == nil {
-			// Set default MaxUnavailable as 1 by default.
-			maxUnavailable := intstr.FromInt(1)
+			// Set default MaxUnavailable as 25% by default.
+			maxUnavailable := intstr.FromString("25%")
```

Comment:
should the default value for kubectl rolling update be also updated , it was last changed from 1 to 0 here https://github.com/kubernetes/kubernetes/issues/18463